backend/src/main/java/hu/flowacademy/worksheet/service/WorksheetService.java

```diff
@@ -121,27 +121,8 @@ public Worksheet setStatusWorksheet(String id, WorksheetStatus status) throws Va
 
     public Worksheet update(String id, Worksheet worksheetReceived) throws ValidationException {
         validateWorksheet(worksheetReceived);
-        Worksheet worksheetToUpdate = worksheetRepository.findById(id).orElseThrow(() -> new ValidationException("No worksheet with the given id " + worksheetReceived.getId()));
-        return addedWorksheet(worksheetReceived, worksheetToUpdate);
-    }
-
-    private Worksheet addedWorksheet(Worksheet worksheetReceived, Worksheet worksheetToUpdate) throws ValidationException {
-        validateWorksheet(worksheetReceived);
-        worksheetToUpdate.setPartner(worksheetReceived.getPartner());
-        worksheetToUpdate.setTypeOfWork(worksheetReceived.getTypeOfWork());
-        worksheetToUpdate.setCustomTypeOfWork(worksheetReceived.getCustomTypeOfWork());
-        worksheetToUpdate.setAssetSettlement(worksheetReceived.getAssetSettlement());
-        worksheetToUpdate.setWorkingTimeAccounting(worksheetReceived.getWorkingTimeAccounting());
-        worksheetToUpdate.setNumberOfEmployees(worksheetReceived.getNumberOfEmployees());
-        worksheetToUpdate.setOverheadHour(worksheetReceived.getOverheadHour());
-        worksheetToUpdate.setDeliveryKm(worksheetReceived.getDeliveryKm());
-        worksheetToUpdate.setAccountSerialNumber(worksheetReceived.getAccountSerialNumber());
-        worksheetToUpdate.setDescription(worksheetReceived.getDescription());
-        worksheetToUpdate.setUsedMaterial(worksheetReceived.getUsedMaterial());
-        worksheetToUpdate.setTypeOfPayment(worksheetReceived.getTypeOfPayment());
-        worksheetToUpdate.setWorkerSignature(worksheetReceived.getWorkerSignature());
-        worksheetToUpdate.setProofOfEmployment(worksheetReceived.getProofOfEmployment());
-        return worksheetRepository.save(worksheetToUpdate);
+        worksheetRepository.findById(id).orElseThrow(() -> new ValidationException("No worksheet with the given id " + id));
+        return worksheetRepository.save(worksheetReceived.toBuilder().id(id).build());
```

Comment:
Ez így nem lesz túl szerencsés, azt beszéltük, hogy a partner-t lekérjük az id alapján. írd vissza az addedWorksheet-et és a partner lekérést valósítsa meg!

Reply:
Azt hittem ezt is irjam át, de akkor visszatettem a régire, aztán majd patrick törődik ott vele